### PR TITLE
Fluent list

### PR DESCRIPTION
**Pull request by JoomFX**
related to: https://github.com/telerik/kendo-themes-private/issues/2

### CODE REVIEW

File: packages/fluent/scss/core/_variables.scss [hunk 1]
```diff
@@ -87,6 +88,19 @@ $kendo-border-radius-sm: calc( #{$kendo-border-radius} / 2 ) !default;
 $kendo-border-radius-md: $kendo-border-radius !default;
 $kendo-border-radius-lg: calc( #{$kendo-border-radius} * 2 ) !default;
 
+// Icons
+$kendo-icon-size: 16px !default;
```
Comment (TeyaVes) [inline packages/fluent/scss/core/_variables.scss:92]: the icon sizes can be taken from the icon component, they do not need to be defined here

File: packages/fluent/scss/core/_variables.scss [hunk 2]
```diff
@@ -87,6 +88,19 @@ $kendo-border-radius-sm: calc( #{$kendo-border-radius} / 2 ) !default;
...
+$kendo-icon-size-lg: 32px !default;
+$kendo-icon-spacing: map-get( $kendo-spacing, 1 ) !default;
+
```
Comment (TeyaVes) [inline packages/fluent/scss/core/_variables.scss:95]: the kendo-icon-spacing var can be added to the vars in the icon component

File: packages/fluent/scss/core/_variables.scss [hunk 3]
```diff
@@ -87,6 +88,19 @@ $kendo-border-radius-sm: calc( #{$kendo-border-radius} / 2 ) !default;
...
+// Metrics
+$kendo-padding-x: 8px !default;
+$kendo-padding-y: 4px !default;
```
Comment (TeyaVes) [inline packages/fluent/scss/core/_variables.scss:98]: use the $kendo-spacing map for these values

File: packages/fluent/scss/core/_variables.scss [hunk 4]
```diff
@@ -87,6 +88,19 @@ $kendo-border-radius-sm: calc( #{$kendo-border-radius} / 2 ) !default;
...
+$kendo-padding-x-sm: math.div($kendo-padding-x, 2) !default;
+$kendo-padding-y-sm: math.div($kendo-padding-y, 2) !default;
+$kendo-padding-x-lg: $kendo-padding-x * 1.5 !default;
+$kendo-padding-y-lg: $kendo-padding-y * 1.5 !default;
```
Comment (TeyaVes) [inline packages/fluent/scss/core/_variables.scss:102]: use calc() when multiplying

File: packages/fluent/scss/list/_layout.scss [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        gap: var( --kendo-icon-spacing, #{$kendo-icon-spacing} );
+        position: relative;
+    }
+
+
+    // List item group label
+    .k-list-item-group-label {
+        padding: 0 .5em;
+        font-size: .75em;
+        position: absolute;
```
Comment (TeyaVes) [inline packages/fluent/scss/list/_layout.scss:142]: do we need variables for these values - padding and font-size ?

File: packages/fluent/scss/list/_theme.scss [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    // List group item
+    .k-list-group-item {
+        @include fill(
+            var(--kendo-list-group-item-text, #{$kendo-list-group-item-text}),
+            var(--kendo-list-group-item-bg, #{$kendo-list-group-item-bg}),
+            var(--kendo-list-group-item-border, #{$kendo-list-group-item-border})
+        );
+        @include box-shadow( var(--kendo-list-group-item-shadow, #{$kendo-list-group-item-shadow}) );
+    }
```
Comment (TeyaVes) [inline packages/fluent/scss/list/_theme.scss:77]: add spacing around all vars

File: packages/fluent/scss/list/index.scss [hunk 7]
```diff
@@ -0,0 +1,25 @@
+// Module meta
+$_kendo-module-meta: (
+    name: "list",
+    dependencies: (
+        "checkbox"
+    )
```
Comment (TeyaVes) [inline packages/fluent/scss/list/index.scss:6]: icon should be a dependency as well

File: packages/fluent/scss/list/_layout.scss [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+        align-items: center;
+        align-content: center;
+        gap: var( --kendo-icon-spacing, #{$kendo-icon-spacing} );
+        position: relative;
+    }
+
+
+    // List item group label
+    .k-list-item-group-label {
+        padding: 0 .5em;
```
Comment (TeyaVes) [inline packages/fluent/scss/list/_layout.scss:140]: in the design it looks like the vertical padding is larger than 0

File: packages/fluent/scss/list/_theme.scss [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+
+    // List header
+    .k-list-group-sticky-header {
+        @include fill(
+            var(--kendo-list-header-text, #{$kendo-list-header-text}),
+            var(--kendo-list-header-bg, #{$kendo-list-header-bg}),
+            var(--kendo-list-header-border, #{$kendo-list-header-border})
+        );
+        @include box-shadow( var(--kendo-list-header-shadow, #{$kendo-list-header-shadow}) );
```
Comment (TeyaVes) [inline packages/fluent/scss/list/_theme.scss:22]: I don't see a box-shadow in the designs, but I might be mistaken

**Review by TeyaVes — CHANGES_REQUESTED** [review]

File: packages/fluent/scss/list/_theme.scss (hunk 9 above)
Comment (JoomFX) [reply packages/fluent/scss/list/_theme.scss:22]: I discussed this with @imasyuko and we agreed that there should be a box-shadow on the sticky header element.

File: packages/fluent/scss/list/_variables.scss [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+
+/// Line height of list header, if no size is set.
+/// @group list
+$kendo-list-header-line-height-sm: null !default;
+$kendo-list-header-line-height-md: null !default;
+$kendo-list-header-line-height-lg: null !default;
+
+/// Font weight of list header.
+/// @group list
+$kendo-list-header-font-weight: $kendo-font-weight-bold !default;
```
Comment (epetrow) [inline packages/fluent/scss/list/_variables.scss:46]: Shouldn't we use var( --kendo-font-weight-bold, normal ) here instead

File: packages/fluent/scss/list/_variables.scss [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+
+/// Font size of list items, if no size is set.
+/// @group list
+$kendo-list-item-font-size-sm: null !default;
+$kendo-list-item-font-size-md: null !default;
+$kendo-list-item-font-size-lg: null !default;
+
+/// Line height of list items, if no size is set.
+/// @group list
+$kendo-list-item-line-height-sm: null !default;
```
Comment (epetrow) [inline packages/fluent/scss/list/_variables.scss:68]: We should refrain from using null values  as fallbacks for css variables and instead use the default value for the given property.

**Review by Juveniel — APPROVED** [review]

**Pull request merged**